### PR TITLE
Kayla Ecker - Caret

### PR DESCRIPTION
**Pull request by kaylaecker**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? |  |
| Describe your API Wrapper.  How did you decide on the methods you created? |  |
| Describe an edge case or failure case test you wrote for your API Wrapper. |  |
| Explain how VCR aids in testing an API. |  |
| What is the Heroku URL of your deployed application? |https://desolate-scrubland-60709.herokuapp.com/  |

### CODE REVIEW

File: lib/edamam_api_wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+class EdamamApiWrapper
+  BASE_URL = "https://api.edamam.com/search?"
+  ID_AND_KEY = "&app_id=#{ENV["EDAMAM_ID"]}&app_key=#{ENV["EDAMAM_KEY"]}"
+
+  def self.search(query)
+    url = BASE_URL + "q=" + query + ID_AND_KEY + "&to=50"
+    data = HTTParty.get(url)
+    recipes = []
+    unless data["hits"].empty?
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb:9]: You should have this check read 
`unless data["hits"].nil?  || data["hits"].empty?`

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+<body>
+  <div>
+    <% if flash[:status] = :failure %>
+    <section class="row status <%= flash[:status] %>">
+      <h3>
+        <%= flash[:message] %>
+      </h3>
+      <%end%>
+    </div>
```
Comment (CheezItMan) [inline app/views/layouts/application.html.erb:20]: Need to fix alignment a bit, plus unnecessary `div` elements.

File: app/views/recipes/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<h1>Here are you results for: "<%= @query %>" </h1>
+<ul>
+  <% @recipes.each do |recipe| %>
+  <li>
+    <div class=name>
+      <%= link_to recipe.name, recipe_path(recipe.name, recipe_id: recipe.id)%>
+    </div>
+    <img src="<%= recipe.image %>", alt="<%= recipe.name %>"/>
```
Comment (CheezItMan) [inline app/views/recipes/index.html.erb:8]: making the image a link would be good as well.

File: lib/edamam_api_wrapper.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    end
+    return recipes
+  end
+
+  def self.find_recipe(id)
+    url = BASE_URL + "r=" + id + ID_AND_KEY
+    data = HTTParty.get(url)
+    recipe = ""
+    unless data.empty?
+      recipe = self.create_recipe(data[0])
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb:23]: You should also check to see if data[0] has values

File: test/controllers/recipes_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      must_respond_with :success
+    end
+
+  end
+
+  describe "index" do
+    it "succeeds when there are recipes for search" do
+      query = "apple"
+      VCR.use_cassette("search") do
+        EdamamApiWrapper.search(query).empty?.must_equal false
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:16]: This is too specific to the wrapper.  Let the controller make the wrapper call, focus on using the controller.

File: test/controllers/recipes_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        get recipes_path(query: query)
+        must_respond_with :success
+      end
+    end
+
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:21]: What about show recipe tests?

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | Good commit messages, more granular commits would be better. |
| Comprehension questions	| **NOOOOOOOOOOOOOOOO**!! |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) | Check |
| Semantic HTML | overall pretty good |
| Errors are reported to the user | Check, nice!  |
| API Wrapper to handle the API requests | Check |
| Controller testing | Yes, but no  show action tests |
| Lib testing | Yes, but your negative cases are failing. |
| Search Functionality | Check |
| List Functionality | Check |
| Show individual item functionality (link to original recipe opens in new tab) | Check |
| **Styling** |  |
| Foundation Styling for responsive layout | The whole site is done in 1-column setups, so it's not really responsive. |
| List View shows 10 items at a time/pagination | Check |
| The app is styled to create an attractive user interface | Check |
| **API Features** | |
| The App attributes Edaman |Check |
| The VCR casettes do not contain the API key | Check |
| **External Resources** |  |
| Link to deployed app on Heroku | Deployed, but it doesn't work! |
|  **Overall** | You need to look at the comprehension questions, and work on styling, as well as practice deploying to Heroku.  See my notes in the code about issues with your tests and wrapper.  You did hit most of the requirements. |
